src/postamble.js

```diff
@@ -44,8 +43,15 @@ if (memoryInitializer) {
   } else if (Module['memoryInitializerPrefixURL']) {
     memoryInitializer = Module['memoryInitializerPrefixURL'] + memoryInitializer;
   }
-  if (ENVIRONMENT_IS_NODE || ENVIRONMENT_IS_SHELL) {
-    var data = Module['readBinary'](memoryInitializer);
+  var data = Module['memoryInitializerBytes'];
+  if (!data) {
+    if (ENVIRONMENT_IS_NODE || ENVIRONMENT_IS_SHELL) {
+      data = Module['readBinary'](memoryInitializer);
+    } else {
+      data = tryParseAsDataURI(memoryInitializer);
```

Comment:
I may have misunderstood before then. But I was hoping we could try to let the browser handle it, and if it failed due to CSP, we would parse it manually as a fallback. Is that not possible?